meteorite_fall.js

```diff
@@ -2,37 +2,52 @@ const canvas = document.getElementById("canvas");
 const context = canvas.getContext("2d");
 const canvasHeight = canvas.height;
 const canvasWidth = canvas.width;
-const meteoriteDisplacement = 5;
+const meteoriteFallSpeed = 5;
+const meteoriteWidth = 10;
+const meteoriteHeight = 40;
 
-let meteoriteXPosition = Math.floor(Math.random() * canvasWidth); 
 let meteoriteYPosition = 0;
 
-let meteorite = {
-    width: 10,
-    height: 40
-};
-
+let meteorites = new Array();
 let player = new Player(canvasWidth / 2, canvasHeight - 10, 10, 10, 10, "blue");
 
 function main() {
-    setInterval(meteoriteFall, 50);
+    fillMeteorites();
+    setInterval(startGame, 50);
 }
 
-function drawMeteorite() {
-    context.beginPath();
-    context.rect(meteoriteXPosition, meteoriteYPosition, meteorite.width, meteorite.height)
-    context.fillStyle = "black";
-    context.fill();
-    context.closePath();
-}
+// main game functionality
+function startGame() {
+    var randomMeteoriteIndex = Math.floor(Math.random() * meteorites.length);
 
-// main game loop
-function meteoriteFall() {
     context.clearRect(0,0, canvasWidth, canvasHeight);
-    drawMeteorite();
-    player.draw(context);
 
-    meteoriteYPosition += meteoriteDisplacement;
+    if (meteorites[randomMeteoriteIndex].isFalling == false) {
+        meteorites[randomMeteoriteIndex].drawMeteorite(context);
```

Comment:
drawing, and making it fall here are useless, as you will do it when you loop through all the meteorites anyway

Reply:
Updated.